### PR TITLE
rephrase the text around dependency diagrams

### PR DESCRIPTION
**Pull request by jhidding**
Slightly rephrases text surrounding code blocks and dependency diagrams in ch1. This should fix issues #11 and #12.

### CODE REVIEW

**Comment by PabRod** [comment]
## Idea to be discussed

I still find the snippets hard to relate to the diagrams. 

What if...

- ... at this step, we use diagrams only and no code snippets at all?

or

- ... we introduce first the diagrams, and the code snippets afterwards?

**Comment by PabRod** [comment]
The latest version, although can certainly be fine-tuned, contains the gist of the changes. 

What do you think, @jhidding, @svenvanderburg?

**Comment by svenvanderburg** [comment]
@PabRod I am planning to take a look again next co-working friday. But let me know if you want to get this through earlier, then I will take a look sooner.

**Comment by PabRod** [comment]
There is no hurry @svenvanderburg. This and next week I'll be busy preparing and teaching the R course.

Thanks!

File: _episodes/01-introduction.md [hunk 1]
```diff
@@ -66,31 +66,72 @@ formalism.
...
+
+### How can parallel computing improve our code execution speed?
+
+Nowadays, most personal computers have 4 or 8 processors (also known as cores). In the diagram above, we can assign each of the three functions to one core, so they can be performed simultaneously.
+
+It may be tempting to think that using three cores instead of one would multiply the execution speed by three. For now, it's ok to use this a first approximation to reality. Later in the course we'll see that things are actually more complicated than that.
+
+## Parallelizable and non-parallelizable tasks
+
+It is important to know that some tasks are fundamentally non-parallelizable, also known as **inherently serial**. An example could be the brute-force computation of the factorial of an integer (example: the factorial of 4, usually written as $4!$ is obtained by multiplying all the integers smaller or equal to $6$, that is: $4! = 4 \cdot 3 \cdot 2 \cdot 1 = 24$ ).
```
Comment (svenvanderburg) [inline _episodes/01-introduction.md:92]: The formulas here do not render properly. (I think the notation with $ only works in R-markdown).

**Review by svenvanderburg — CHANGES_REQUESTED** [review]
I really like these changes! It's really clear now what the difference is between parallel and serial tasks. Just had one comment (that should be fixed though).

File: _episodes/01-introduction.md (hunk 1 above)
Comment (svenvanderburg) [reply _episodes/01-introduction.md]: I've committed a fix (but I am no expert in this, I just followed https://stackoverflow.com/a/47798853/4711118

Comment (PabRod) [reply _episodes/01-introduction.md]: If I remember properly, we had a similar issue before. We fixed it by adding [this snippet](https://github.com/escience-academy/lesson-template/commit/fcb625142bdbf1b4c746d99640de21a6e07e8a5c).

Comment (PabRod) [reply _episodes/01-introduction.md]: Applied the snippet in commit 7c94d59608b1a15c3af4ad4c016c56cbf7d27ce3.

**Review by PabRod — APPROVED** [review]
These changes have been implemented. Thanks!

**Review by PabRod — APPROVED** [review]
This looks good enough. The text improvement is a potential never-ending story.

**Pull request merged**